src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -274,13 +351,129 @@ internal static byte[][] SegmentedEncodeNamedBitList(byte[] bigEndianBytes, int
                 dataSegment
             };
         }
-        
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of an octet string (byte array).
+        /// </summary>
+        /// <param name="data">The data to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeOctetString(byte[] data)
+        {
+            Debug.Assert(data != null);
+
+            // Because this is not currently public API the data array is not being cloned.
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.OctetString }, 
+                EncodeLength(data.Length),
+                data,
+            };
+        }
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of an object identifier (Oid).
+        /// </summary>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeOid(Oid oid)
+        {
+            if (oid == null)
```

Comment:
That’s fine – that specific null check was the one that I was commenting on. Stuff that the caller can’t reasonably check is fine to throw on.